lib/discourse_js_processor.rb

```diff
@@ -163,6 +179,58 @@ def module_name(root_path, logical_path)
       # We need to strip the app subdirectory to replicate how ember-cli works.
       path || logical_path&.gsub('app/', '')&.gsub('addon/', '')&.gsub('admin/addon', 'admin')
     end
+  end
 
+  class BabelHelper
+    PLUGINS = [
+      'transform-literals',
+      'transform-named-capturing-groups-regex',
+      'transform-sticky-regex',
+      'transform-unicode-regex',
+      'transform-for-of',
+      'transform-function-name',
+      'transform-spread',
+      'transform-typeof-symbol',
+      'transform-computed-properties',
+      'transform-destructuring',
+      'transform-duplicate-keys',
+      'transform-arrow-functions',
+      'transform-block-scoped-functions',
+      'transform-block-scoping',
+      'transform-object-super',
+      'transform-parameters',
+      'transform-shorthand-properties',
+      'transform-dotall-regex',
+      'proposal-json-strings',
+      'proposal-nullish-coalescing-operator',
+      'proposal-logical-assignment-operators',
+      'proposal-numeric-separator',
+      'proposal-optional-catch-binding',
+      'proposal-object-rest-spread',
+      'proposal-optional-chaining',
+      'proposal-unicode-property-regex',
+      "['proposal-decorators', { legacy: true }]"
+    ]
+
+    def self.build(source, plugins = [], options = {})
+      options[:ast] ||= false
+
+      options_string = options.map do |k, v|
+        v.is_a?(String) ? "#{k}: '#{v}'" : "#{k}: #{v}"
+      end.join(', ')
+
+      plugins_string = (PLUGINS + plugins).map do |plugin|
+        # eg: "['proposal-decorators', { legacy: true }]"
+        # should be treated as an array in the generated script and not a string
+        # same for exports
+        if plugin.start_with?('[') || plugin.start_with?('exports')
```

Comment:
This feels like a bit of a hack to me. Rather than checking if the value is an array, could you include it in the array without a string encoding like this:

```ruby
[
  ['proposal-decorators', { legacy: true }]
]
```

Then you could call `to_json` on each member of the array to have it encoded propery?

`exports` would need an exception but you could add another parameter type for that maybe?

Reply:
Yes that was my feeling, I couldn't get everything to work in one clean way, so I would rather have a hack at one place solving all cases. I will try to find something better tomorrow.